components/scream/src/physics/shoc/shoc_functions_f90.cpp

```diff
@@ -3188,9 +3188,77 @@ void pblintd_check_pblh_f(Int shcol, Int nlev, Int nlevi, Int npbl, Real* z, Rea
   ekat::device_to_host({pblh}, shcol, out_1d_views);
 }
 
-void pblintd_f(Int shcol, Int nlev, Int nlevi, Real* z, Real* zi, Real* thl, Real* ql, Real* q, Real* u, Real* v, Real* ustar, Real* obklen, Real* kbfs, Real* cldn, Real* pblh)
+void pblintd_f(Int shcol, Int nlev, Int nlevi, Real* z, Real* zi, Real* thl, Real* ql, Real* q, 
+              Real* u, Real* v, Real* ustar, Real* obklen, Real* kbfs, Real* cldn, Real* pblh)
 {
-  // TODO
+  using SHOC       = Functions<Real, DefaultDevice>;
+  using Spack      = typename SHOC::Spack;
+  using Scalar     = typename SHOC::Scalar;
+  using Pack1      = typename ekat::Pack<Real, 1>;
+  using view_1d    = typename SHOC::view_1d<Pack1>;
+  using view_2d    = typename SHOC::view_2d<Spack>;
+  using ExeSpace   = typename SHOC::KT::ExeSpace;
+  using MemberType = typename SHOC::MemberType;
+
+  std::vector<view_2d> views_2d(8);
+  ekat::host_to_device({z, zi, thl, ql, q, u, v, cldn}, shcol, nlev, views_2d, true);
+
+  view_2d z_2d    (views_2d[0]),
+          zi_2d   (views_2d[1]),
+          thl_2d  (views_2d[2]),
+          ql_2d   (views_2d[3]),
+          q_2d    (views_2d[4]),
+          u_2d    (views_2d[5]),
+          v_2d    (views_2d[6]),
+          cldn_2d (views_2d[7]);
+
+  std::vector<view_1d> views_1d(3);
+  ekat::host_to_device({ustar, obklen, kbfs}, shcol, views_1d);
+  view_1d ustar_1d   (views_1d[0]),
+          obklen_1d  (views_1d[1]),
+          kbfs_1d    (views_1d[2]);
+
+  SHOC::view_1d<bool> check_1d("check", shcol);
+  const auto host_check_1d = Kokkos::create_mirror_view(check_1d);
+  Kokkos::deep_copy(check_1d, host_check_1d);
```

Comment:
Use `ekat::host_to_device` for bools

Reply:
sorry, it is a local variable, so we don't need this one, I deleted it though.